core/src/com/unciv/logic/city/City.kt

```diff
@@ -412,9 +417,20 @@ class City : IsPartOfGameInfoSerialization {
         }
         if (!manualSpecialists)
             population.specialistAllocations.clear()
+        updateCitizens = false
         population.autoAssignPopulation()
     }
 
+    /** Apply worked tiles optimization (aka CityFocus) -
+     *  immediately for a human player whoes turn it is (interactive),
+     *  or deferred to the next startTurn while nextTurn is running (for AI)
+     *  @see reassignPopulation
+     */
+    fun reassignPopulationDeferred() {
+        // TODO - is this the best (or even correct) way to detect "interactive" UI calls?
+        if (GUI.isMyTurn() && GUI.getViewingPlayer() == civ) reassignPopulation()
```

Comment:
Absolutely not - this information is in the *game data*, so we can check city.civ.gameInfo for this info without addressing the GUI at all

Reply:
Why would you want to dever stuff on start turn and end turn? Performance reasons? Reassign population is not very heavy IIRC